Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -471,21 +475,32 @@ inline void ClearPath() {
     }
 
     inline bool ShouldDraw() {
-        return CurrentGState().ShouldDraw();
+        return FAILED(_firstErrorHr) || CurrentGState().ShouldDraw();
     }
 
     inline void PushBeginDraw() {
-        if ((_beginEndDrawDepth)++ == 0) {
+        if (!FAILED(_firstErrorHr) && (_beginEndDrawDepth)++ == 0) {
             deviceContext->BeginDraw();
         }
     }
 
     inline HRESULT PopEndDraw() {
+        HRESULT hr = S_OK;
         if (--(_beginEndDrawDepth) == 0) {
-            RETURN_IF_FAILED(deviceContext->EndDraw());
+            hr = deviceContext->EndDraw();
+            if (_useEnhancedErrorHandling && !FAILED(_firstErrorHr) && FAILED(hr)) {
+                // If we haven't yet stored an error, and we're about to return an error (not S_OK), store it.
+                _firstErrorHr = hr;
+                hr = S_OK;
+            }
         }
-        return S_OK;
+        return hr;
+    }
+
+    inline void EnableEnhancedErrorHandling() {
```

Comment:
@aballway no; external bitmap context consumers are not enlightened as to error handling.